Kudu.Core/Infrastructure/DeploymentLockFile.cs

```diff
@@ -25,6 +36,29 @@ public IRepositoryFactory RepositoryFactory
 
         protected override void OnLockAcquired()
         {
+
+            try
+            {
+                // Create Sentinel file for DWAS to check
+                // DWAS will check for presence of this file incase a an app setting based recycle needs to be performed in middle of deployment
+                // If this is present, DWAS will postpone the recycle so that deployment goes through first
+                if (!String.IsNullOrEmpty(siteRoot))
+                {
+                    FileSystemHelpers.CreateDirectory(Path.Combine(siteRoot, @"ShutdownSentinel"));
+                    string sentinelPath = Path.Combine(siteRoot, @"ShutdownSentinel\Sentinel.txt");
+                    var file = FileSystemHelpers.CreateFile(sentinelPath);
```

Comment:
What happens if file already exists?  Will throw?   I would do this .. if file not exists, create one.   Then update the time.